openfl/callbacks/callback_list.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2020-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+from typing import List
+
+from openfl.callbacks.callback import Callback
+from openfl.callbacks.memory_profiler import MemoryProfiler
+
+
+class CallbackList(Callback):
+    """An ensemble of callbacks.
+
+    This class allows multiple callbacks to be used together, by sequentially
+    calling each callback's respective methods.
+
+    Attributes:
+        callbacks: A list of `openfl.callbacks.Callback` instances.
+        add_memory_profiler: If True, adds a `MemoryProfiler` callback to the list.
+        tensor_db: Optional `TensorDB` instance of the respective participant.
+            If provided, callbacks can access TensorDB for various actions.
+        params: Additional parameters saved for use within the callbacks.
+    """
+
+    def __init__(
+        self,
+        callbacks: List[Callback],
+        add_memory_profiler=False,
+        tensor_db=None,
+        **params,
+    ):
+        super().__init__()
+        self.callbacks = _flatten(callbacks) if callbacks else []
+
+        self._add_default_callbacks(add_memory_profiler)
+
+        self.set_tensor_db(tensor_db)
+        self.set_params(params)
+
+    def set_params(self, params):
+        self.params = params
+        if params:
+            for callback in self.callbacks:
+                callback.set_params(params)
+
+    def set_tensor_db(self, tensor_db):
+        self.tensor_db = tensor_db
+        if tensor_db:
+            for callback in self.callbacks:
+                callback.set_tensor_db(tensor_db)
+
+    def _add_default_callbacks(self, add_memory_profiler):
+        self._memory_profiler = None
+        for cb in self.callbacks:
+            if isinstance(cb, MemoryProfiler):
+                self._memory_profiler = cb
```

Comment:
we can put break after this.

Reply:
Agree, for the code today. 

However, there will be more default callbacks coming in. Take this example of default `MemoryProfiler` and `MetricLogger`:

```python
for cb in self.callbacks:
  if isinstance(cb, MemoryProfiler):
    self._memory_profiler = cb
  elif isinstance(cb, MetricLogger):
    self._metric_logger = cb
```

A break exits the loop when we don't intend to. The downside of not having `break`, is that callbacks list will only assign the 'latest' seen callback instead of the 'first' seen callback.